frontend/src/components/sessions/data-visualization/StatisticsSection.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { Flex, Grid, Text } from "@chakra-ui/react";
+
+import StatisticCard, { type StatisticCardVariant } from "./StatisticCard";
+
+export type StatisticsConfig = {
+  [key: string]: {
+    title: string;
+    formatValue: (value: number) => string;
+    variant?: StatisticCardVariant;
+    fallbackValue?: string;
+  };
+};
+
+type Statistics<T> = {
+  [key in keyof T]?: number;
+};
+
+const getKeys = <T,>(obj: StatisticsConfig) =>
+  Object.keys(obj) as (keyof T & string)[];
+
+export const NUMBER_FORMAT = (value: number) => value.toFixed(0);
+export const PERCENTAGE_FORMAT = (value: number) => `${NUMBER_FORMAT(value)}%`;
+export const NTH_FORMAT = (value: number) => {
+  const lastDigit = value % 10;
+  let suffix = "th";
+  if (lastDigit === 1) suffix = "st";
+  if (lastDigit === 2) suffix = "nd";
+  if (lastDigit === 3) suffix = "rd";
+  return `${NUMBER_FORMAT(value)}${suffix}`;
+};
```

Comment:
thoughts on moving `NUMBER_FORMAT`, `PERCENTAGE_FORMAT`, and `NTH_FORMAT` to a separate utils file?

Reply:
sure